app/src/main/java/chat/rocket/android/chatroom/adapter/MessageViewHolder.kt

```diff
@@ -77,10 +80,28 @@ class MessageViewHolder(
                 read_receipt_view.isVisible = true
             }
 
-            image_avatar.setOnClickListener {
-                data.message.sender?.id?.let { userId ->
-                    avatarListener(userId)
+            if (!groupMessage) {
+                layout_avatar.isVisible = true
+                message_header.isVisible = true
+
+                (text_content.layoutParams as ViewGroup.MarginLayoutParams).apply {
+                    marginStart = 16.toPx.toInt()
                 }
+
+                image_avatar.setOnClickListener {
+                    data.message.sender?.id?.let { userId ->
+                        avatarListener(userId)
+                    }
+                }
+            } else {
+                layout_avatar.isGone = true
+                message_header.isGone = true
```

Comment:
what is the difference between isGone and isVisible? Shouldn't this just be controlled by a single boolean variable set to true or false?

Reply:
When view is invisible, it is not shown but it still takes space for layout purposes.

But if view visibility is Gone then it is not shown and also it doesn't take any space for layout purposes